README.md

```diff
@@ -582,6 +582,32 @@ func main() {
         // all is good, use result
     }
 }
+```
+Here's an example:
+```go
+type number struct {
+	n int
+}
+
+// define a sqrt function on "number" type that returns
+// a float64 and error value based on the input
+func (N number) sqrt() (val float64, err error) {
```

Comment:
Please merge this example with the one above by replacing the `doStuff` function.

Also, this struct is not really for this example. Let's define it only as a single function.

Thanks